javascript/src/main/components/BasicCourseSearch/CourseSearchFormQtrDeptOnly.js

```diff
@@ -1,52 +1,56 @@
+import SelectQuarter from "main/components/BasicCourseSearch/SelectQuarter";
 
 const CourseSearchFormQtrDeptOnly = ({ setCourseJSON, fetchJSON }) => {
-    const [quarter, setQuarter] = useState("20212");
-    const [department, setDepartment] = useState("CMPSC");
-    const { addToast } = useToasts()
-    
-    const handleSubmit = (event) => {
-        event.preventDefault();
-        fetchJSON(event, { quarter, department }).then((courseJSON) => {
-            if(courseJSON.total === 0){
-                addToast("There are no courses that match the requested criteria.", { appearance: "error" });
-            }
-            setCourseJSON(courseJSON);
-        });
-    };
+	const quarters = quarterRange("20084", "20214");
+	const [quarter, setQuarter] = useState(quarters[0].yyyyq);
+	const [department, setDepartment] = useState("CMPSC");
+	const { addToast } = useToasts();
 
-    const handleQuarterOnChange = (event) => {
-        setQuarter(event.target.value);
-    };
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		fetchJSON(event, { quarter, department }).then((courseJSON) => {
+			if (courseJSON.total === 0) {
+				addToast("There are no courses that match the requested criteria.", {
+					appearance: "error",
+				});
+			}
+			setCourseJSON(courseJSON);
+		});
+	};
 
-    const handleDepartmentOnChange = (event) => {
-        setDepartment(event.target.value);
-    };
+	const handleDepartmentOnChange = (event) => {
+		setDepartment(event.target.value);
+	};
 
-    return (
-        <Form onSubmit={handleSubmit}>
-            <Form.Group controlId="BasicSearch.Quarter">
-                <Form.Label>Quarter</Form.Label>
-                <Form.Control as="select" onChange={handleQuarterOnChange} value={quarter}  >
-                    <option value="20212">S21</option>
-                    <option value="20211">W21</option>
-                    <option value="20204">F20</option>
-                </Form.Control>
-            </Form.Group>
-            <Form.Group controlId="BasicSearch.Department">
-                <Form.Label>Department</Form.Label>
-                <Form.Control as="select" onChange={handleDepartmentOnChange} value={department}>
-                    <option>CMPSC</option>
-                    <option>MATH</option>
-                </Form.Control>
-            </Form.Group>
-            <Button variant="primary" type="submit">
-                Submit
-            </Button>
-        </Form>
-    );
+	return (
+		<Form onSubmit={handleSubmit}>
+			<SelectQuarter
+				quarters={quarters}
+				quarter={quarter}
+				setQuarter={setQuarter}
+				controlId={"BasicSearch.Quarter"}
+				label={"Quarter"}
+			/>
+			<Form.Group controlId="BasicSearch.Department">
+				<Form.Label>Department</Form.Label>
+				<Form.Control
+					as="select"
+					onChange={handleDepartmentOnChange}
+					value={department}
+				>
+					<option>CMPSC</option>
+					<option>MATH</option>
```

Comment:
This isn't a blocker for this particular PR.  But I'd suggest, if there isn't already a story for this, making a story to replace this hard coded list with a usage of the `SelectSubject` component.

But, again, that's for another story (and that story may already exist?)